pkg/cri/sbserver/podsandbox/controller.go

```diff
@@ -149,7 +165,7 @@ func (c *Controller) waitSandboxExit(ctx context.Context, id string, exitCh <-ch
 }
 
 // handleSandboxExit handles TaskExit event for sandbox.
-func handleSandboxExit(ctx context.Context, sb sandboxstore.Sandbox) error {
+func handleSandboxExit(ctx context.Context, e *eventtypes.TaskExit, sb sandboxstore.Sandbox) error {
```

Comment:
suggest to use `handleSandboxExit(context.Context, sandboxstore.Sandbox, TaskExit)`

and since the TaskEvent is designed for shim.Task, do you need to extend the TaskExit with sandboxID here?
cc @mxpv @mikebrow

Reply:
Good question, in order to keep the responsibility single, `TaskEvent` should not be used here in theory, but `TaskEvent` is currently used elsewhere, e.g.a
- in `startSandboxExitMonitor()` 
https://github.com/containerd/containerd/blob/271628971fb46a8e598dc2c7b24d43c78092a5d3/pkg/cri/server/events.go#L122

- in `RunPodSandbox()` https://github.com/containerd/containerd/blob/271628971fb46a8e598dc2c7b24d43c78092a5d3/pkg/cri/sbserver/sandbox_run.go#L215

About this problem, I think it can be solved in two steps:
1. In the short term, `TaskEvent` is still used, consistent with other places
2. In the long run, we can add a `SandboxEvent` designed for `sandbox`, and then make a unified replacement.

Reply:
@fuweid @lengrongfu would it be ok to leave a TODO for this and get this PR in?

Reply:
Add TODO issue https://github.com/containerd/containerd/issues/7548